### PR TITLE
DLP: Added sample for Deidentify data using redact

### PR DESCRIPTION
**Pull request by jainamtrivedi-crest**
- Added sample for [deidentify data using redact with matched input values](https://cloud.google.com/dlp/docs/samples/dlp-deidentify-redact) and added unit tests

### CODE REVIEW

File: dlp/api/Snippets.Tests/DeidentifyDataUsingRedactWithMatchedInputValuesTests.cs [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+        [Fact]
+        public void TestDeidentifyRedact()
+        {
+            var text = "My phone number is 4359916732.";
+            var infoTypes = new InfoType[] { new InfoType { Name = "PHONE_NUMBER" } };
+            var expectedText = "My phone number is .";
+            var result = DeidentifyDataUsingRedactWithMatchedInputValues.Deidentify(Fixture.ProjectId, text, infoTypes);
+            Assert.Equal(expectedText, result.Item.Value);
+            Assert.NotEqual(text, result.Item.Value);
```
Comment (annadata) [inline dlp/api/Snippets.Tests/DeidentifyDataUsingRedactWithMatchedInputValuesTests.cs:36]: This looks like a redundant assert. Pl remove.

Comment (jainamtrivedi-crest) [reply dlp/api/Snippets.Tests/DeidentifyDataUsingRedactWithMatchedInputValuesTests.cs]: Removed

**Review by annadata — APPROVED** [review]

**Review by amanda-tarafa — CHANGES_REQUESTED** [review]
Mostly fine, just a couple of change requests.

Importantly, please reabse into a single commit.

File: dlp/api/Snippets/DeidentifyDataUsingRedactWithMatchedInputValues.cs [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+        var dlp = DlpServiceClient.Create();
+
+        // Construct inspect config.
+        var inspectConfig = new InspectConfig
+        {
+            InfoTypes = { infoTypes ?? new InfoType[] { new InfoType { Name = "EMAIL_ADDRESS" } } },
+        };
+
+        // Construct redact config.
+        var redactConfig = new RedactConfig { };
```
[the remark this answers is not in the document]
Comment (jainamtrivedi-crest) [reply dlp/api/Snippets/DeidentifyDataUsingRedactWithMatchedInputValues.cs]: Done

File: dlp/api/Snippets.Tests/DeidentifyDataUsingRedactWithMatchedInputValuesTests.cs [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+// limitations under the License
+
+using Google.Cloud.Dlp.V2;
+using Xunit;
+
+namespace GoogleCloudSamples
+{
+    public class DeidentifyDataUsingRedactWithMatchedInputValuesTests : IClassFixture<DlpTestFixture>
+    {
+        private DlpTestFixture Fixture { get; }
```
[the remark this answers is not in the document]
Comment (jainamtrivedi-crest) [reply dlp/api/Snippets.Tests/DeidentifyDataUsingRedactWithMatchedInputValuesTests.cs]: Done.

File: dlp/api/Snippets.Tests/DeidentifyDataUsingRedactWithMatchedInputValuesTests.cs [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+        public DeidentifyDataUsingRedactWithMatchedInputValuesTests(DlpTestFixture fixture)
+        {
+            Fixture = fixture;
+        }
```
[the remark this answers is not in the document]
Comment (jainamtrivedi-crest) [reply dlp/api/Snippets.Tests/DeidentifyDataUsingRedactWithMatchedInputValuesTests.cs]: Done

**Pull request merged**